src/loaders/gtdb/compute_genome_attribs.py

```diff
@@ -0,0 +1,369 @@
+                          help='Number of threads for each program execution. (default: 32)')
+    optional.add_argument('--node_id', type=str, default=str(uuid.uuid4()),
+                          help='node ID for running job')
+    optional.add_argument('--debug', action='store_true',
+                          help='Debug mode.')
+    optional.add_argument('--genome_id_file', type=argparse.FileType('r'),
+                          help="tab separated file containing genome ids for the running job")
+    args = parser.parse_args()
+
+    (tools,
+     load_ver,
+     source_data_dir,
+     kbase_collection,
+     root_dir,
+     threads,
+     program_threads,
+     debug,
+     genome_id_file,
+     node_id) = (args.tools, args.load_ver, args.source_data_dir,
+                 args.kbase_collection, args.root_dir, args.threads, args.program_threads,
+                 args.debug, args.genome_id_file, args.node_id)
+
+    # get all genome ids (folder name) from source data directory
+    all_genome_ids = [path for path in os.listdir(source_data_dir) if
+                      os.path.isdir(os.path.join(source_data_dir, path))]
+
+    if not all_genome_ids:
+        raise ValueError('Please download genome files first.')
+
+    # get genome ids for the running job
+    if genome_id_file:
+        # parse genome ids from tab separated file
+        with genome_id_file:
+            df = pd.read_csv(genome_id_file, sep='\t')
+            try:
+                genome_ids = df['genome_id']
+            except KeyError:
+                raise ValueError('Please ensure genome_id column exists in the genome id file.')
+
+        if not set(all_genome_ids) >= set(genome_ids):
+            raise ValueError(f'Genome files for {set(genome_ids) - set(all_genome_ids)} are not downloaded yet')
+    else:
+        # executing all genomes
+        genome_ids = all_genome_ids
+
+    total_count = len(genome_ids)
+
+    if not threads:
+        threads = max(int(multiprocessing.cpu_count() * min(SYSTEM_UTILIZATION, 1)), 1)
+    threads = max(1, threads)
+
+    for tool in tools:
+        # execute each tooling in series
+        try:
+            comp_ops = getattr(sys.modules[__name__], tool)
+        except AttributeError as e:
+            raise ValueError(
+                f'Please implement method for: [{tool}]\n'
+                f'NOTE: Method name should be exactly the same as the tool name') from e
+
+        # place computed results to COLLECTION_DATA_DIR directory
+        work_dir = os.path.join(root_dir, COLLECTION_DATA_DIR, kbase_collection,
+                                load_ver, tool)
+
+        start = time.time()
+        print(f"Start executing {tool} with {threads} threads")
+        if tool in SERIES_TOOLS:
+            failed_ids = comp_ops(genome_ids, work_dir, source_data_dir, debug, threads)
+        else:
+            # call tool execution in parallel
+            batch_size = max(math.floor(threads / program_threads), 1)
```

Comment:
is `batch_count` a more descriptive name here? I think that's what this variable is but `size` is making me second guess what's intended

Reply:
or `num_batches` or something

Reply:
👍

Reply:
:+1: